### PR TITLE
exception/policy: add stats counters - v6

### PR DESCRIPTION
**Pull request by jufajardini**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/5816

Hiding 0 stats counters:
https://redmine.openinfosecfoundation.org/issues/5976

Previous PR: https://github.com/OISF/suricata/pull/10527

Describe changes:
- make logging of stats that are zero optional (independent commit) - with this change, we go back to not being able to disable zero valued counters just for exception policies, though
- extract most of stats counts registration for exception policies into functions
- minor yaml update for the wording around Suricata version
- yaml - explicitly indicate the settings that are related to exception policies (to help with user awareness)
- increase stats for flow memcap exception policy counters when we actually apply such policies
- wrap check for ThreadVars NULL within unittests context, when possible
- rename config setting names for exception policy stats

TODO?:

better align stats.log - as the new counters size character is longer than what we currently have in the stats

### Provide values to any of the below to override the defaults.

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1742

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 19916

File: src/app-layer.c [hunk 1]
```diff
@@ -159,6 +190,25 @@ void AppLayerIncInternalErrorCounter(ThreadVars *tv, Flow *f)
     }
 }
 
+static void AppLayerIncrErrorExcPolicyCounter(ThreadVars *tv, Flow *f, enum ExceptionPolicy policy)
+{
+#ifdef UNITTESTS
+    if (likely(tv == NULL)) {
```
Comment (jufajardini) [inline src/app-layer.c:196]: Do not use likely annotation here.

File: src/flow-hash.c [hunk 2]
```diff
@@ -641,9 +641,24 @@ static inline Flow *FlowSpareSync(ThreadVars *tv, FlowLookupStruct *fls,
     return f;
 }
 
-static inline void NoFlowHandleIPS(Packet *p)
+static void FlowExceptionPolicyStatsIncr(
+        ThreadVars *tv, FlowLookupStruct *fls, enum ExceptionPolicy policy)
+{
+#ifdef UNITTESTS
+    if (likely(tv == NULL)) {
+        return;
```
Comment (jufajardini) [inline src/flow-hash.c:649]: Do not use likely here.

**Comment by jufajardini** [comment]
Replaced by: https://github.com/OISF/suricata/pull/10779

**Pull request closed without merging**